cmake/find/curl.cmake

```diff
@@ -0,0 +1,40 @@
+# curl is required by s3/sentry
+option(ENABLE_CURL "Enable curl" ${ENABLE_LIBRARIES})
+
+set (CURL_LIBRARY ${ClickHouse_SOURCE_DIR}/contrib/curl/lib)
+set (CURL_INCLUDE_DIR ${ClickHouse_SOURCE_DIR}/contrib/curl/include)
+
+if (ENABLE_CURL)
+
+option(USE_INTERNAL_CURL_LIBRARY "Set to FALSE to use system curl library instead of bundled" ${NOT_UNBUNDLED})
+
+if (NOT EXISTS "${ClickHouse_SOURCE_DIR}/contrib/curl-cmake/CMakeLists.txt")
+    if (USE_INTERNAL_CURL_LIBRARY)
+        message(WARNING "submodule contrib/curl-cmake is missing. to fix try run: \n git submodule update --init --recursive")
+    endif()
+    set(USE_INTERNAL_CURL_LIBRARY 0)
+    set(MISSING_INTERNAL_CURL_LIBRARY 1)
+endif()
+
+set (CURL_USE_STATIC_LIBS ${USE_STATIC_LIBRARIES})
+
+if (NOT USE_INTERNAL_CURL_LIBRARY)
+    find_package (CURL)
+endif ()
+
+if (NOT CURL_FOUND)
```

Comment:
FWIW because of this fallback unbundled build works even w/o docker image update